source/Engine.cpp

```diff
@@ -992,8 +982,9 @@ void Engine::Draw() const
 	const Interface *hud = GameData::Interfaces().Get("hud");
 
 	// Draw any active planet labels.
-	for(const PlanetLabel &label : labels)
-		label.Draw();
+	if(Preferences::Has("Show planet labels"))
+		for(const PlanetLabel &label : labels)
+			label.Draw();
```

Comment:
So we're always calculating the planet label locations, just not showing them if the preference is set as such? Why not just not calculate their locations, as it seems was done before?

Reply:
You mean, why not skip calculating the locations if we're not drawing them anyway?  Well, I dunno when the preference changes, and it could be awkward if it changes between ::Step and ::Draw.

On the other hand, I supose you could be asking why not just calculate and draw at the same time (or not, depending on pref)?  It just seemed like a lot of stuff made the same split.  Perhaps the center and/or zoom could change between ::Step and ::Zoom?  I notice that while a few things use zoom in ::Draw, almost nothing uses center (just the background).  I don't have an good understanding of how the threads interact.

Also, the updating is pretty minimal.

Reply:
Ah, the previous version removed the labels entirely if the preference was off, so it wasn't an issue.  Which it could get away with, as it created them anew each ::Step.  The new way, we create them once at ::EnterSystem.  So, if they're no updated at ::Step, they could be misdrawn at ::Draw - we can't just delete them the way the old version did it.

Reply:
I don't imagine this to be some performance hog, so it's fine to keep it as is. Was just curious.